app/services/user-theme.js

```diff
@@ -0,0 +1,92 @@
+import Ember from 'ember';
+import C from 'ui/utils/constants';
+
+const AUTO_UPDATE_TIMER = 1800000; // 30min
+
+export default Ember.Service.extend({
+  prefs : Ember.inject.service(),
+
+  defaultUiTheme : null,
+
+  themeObserver: function() {
+    if (this.get('defaultUiTheme') === 'ui-auto') {
+      this.setAutoUpdate();
+    }
+  }.observes('defaultUiTheme'),
+
+  setupTheme: function() {
+    var userTheme    = this.get(`prefs.${C.PREFS.DEFAULT_UI_THEME}`);
+    var defaultTheme = this.get('defaultUiTheme');
+
+    // does the uitheme exist in prefs
+    if (userTheme) {
+
+      // dooes the user pref'd theme match the default theme from local storage
+      if (userTheme !== defaultTheme) {
+
+        if (defaultTheme !== 'ui-auto') {
+          this.set(`prefs.${C.PREFS.DEFAULT_UI_THEME}`, userTheme);
+        } else {
+          this.setAutoUpdate();
+        }
+
+        window.localStorage.setItem('defaultUiTheme', userTheme);
```

Comment:
`set('prefs...)` and `localStorage.setItem` are setting the same key twice with slightly different values (prefs json-encodes it)?